alert/store.go

```diff
@@ -293,18 +300,37 @@ func (s *Store) EscalateMany(ctx context.Context, alertIDs []int) ([]int, error)
 		return nil, nil
 	}
 
-	err = validate.Range("AlertIDs", len(alertIDs), 1, maxBatch)
+	tx, err := s.db.BeginTx(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
+	defer tx.Rollback()
 
-	ids := sqlutil.IntArray(alertIDs)
+	// return error if any services for each alert are in maintenance mode
+	for _, alertID := range alertIDs {
+		var maintExpiresAt sql.NullTime
+		var svcID string
+		err := tx.StmtContext(ctx, s.getServiceID).QueryRowContext(ctx, alertID).Scan(&svcID)
+		if err != nil {
+			return nil, err
+		}
 
-	tx, err := s.db.BeginTx(ctx, nil)
+		err = tx.StmtContext(ctx, s.maintenanceMode).QueryRowContext(ctx, svcID).Scan(&maintExpiresAt)
+		if err != nil {
+			return nil, err
+		}
+
+		if maintExpiresAt.Valid {
+			return nil, validation.NewGenericError("escalate alert: in maintenance mode")
+		}
+	}
```

Comment:
Might be able to simplify this, but in any case, should happen _after_ the `lockAlertSvc` line; otherwise, we have a race condition from the moment we check the value and the time we actually grab the lock.